### PR TITLE
multichannel to channel_axis (1 of 6): features and draw

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This is the first of a series of 6 PRs completing the `multichannel` to `channel_axis` migration. I am breaking these up into separate modules to keep the size more manageable to review, but the PRs should be reviewed in order since each is based on the previous one.

The decorators used here were introduced in #5228, but an additional check for positional use of the `multichannel` argument has been added to `deprecate_multichannel_kwarg`.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/_shared/utils.py [hunk 1]
```diff
@@ -171,26 +171,46 @@ class deprecate_multichannel_kwarg(deprecate_kwarg):
...
             kwarg_mapping={'multichannel': 'channel_axis'},
             warning_msg=None,
             removed_version=removed_version)
+        self.position = multichannel_position
 
     def __call__(self, func):
         @functools.wraps(func)
         def fixed_func(*args, **kwargs):
 
+            if self.position is not None and len(args) > self.position:
```
Comment (jni) [inline skimage/_shared/utils.py:185]: Nice catch!

**Review by jni — COMMENTED** [review]
Great, thank you @grlee77!

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
@scikit-image/core anyone else have a chance to review this? This is one of the most important changes in moving to skimage 1.0.

File: skimage/feature/_basic_features.py [hunk 2]
```diff
@@ -139,22 +144,32 @@ def multiscale_basic_features(
...
+        to channels.
 
+        .. versionadded:: 0.19
+           ``channel_axis`` was added in 0.19.
 
     Returns
     -------
     features : np.ndarray
-        Array of shape ``image.shape + (n_features,)``
+        Array of shape ``image.shape + (n_features,)``. When channel_axis is
```
Comment (stefanv) [inline skimage/feature/_basic_features.py:158]:
```suggestion
        Array of shape ``image.shape + (n_features,)``. When `channel_axis` is
```

File: skimage/feature/_basic_features.py [hunk 3]
```diff
@@ -139,22 +144,32 @@ def multiscale_basic_features(
...
     """
     if not any([intensity, edges, texture]):
         raise ValueError(
-                "At least one of ``intensity``, ``edges`` or ``textures``"
-                "must be True for features to be computed."
-                )
-    if image.ndim < 3:
-        multichannel = False
-    if not multichannel:
+            "At least one of ``intensity``, ``edges`` or ``textures``"
```
Comment (stefanv) [inline skimage/feature/_basic_features.py:164]:
```suggestion
            "At least one of `intensity`, `edges` or `textures`"
```

File: skimage/feature/_hog.py [hunk 4]
```diff
@@ -96,7 +99,15 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
...
-        otherwise as spatial.
+        otherwise as spatial. This argument is deprecated: specify
+        `channel_axis` instead.
+    channel_axis : int or None, optional
+        If None, the image is assumed to be a grayscale (single channel) image.
+        Otherwise, this parameter indicates which axis of the array corresponds
+        to channels.
+
+        .. versionadded:: 0.19
+           ``channel_axis`` was added in 0.19.
```
Comment (stefanv) [inline skimage/feature/_hog.py:110]:
```suggestion
           `channel_axis` was added in 0.19.
```

File: skimage/feature/tests/test_basic_features.py [hunk 5]
```diff
@@ -1,30 +1,70 @@
...
 
+
 @pytest.mark.parametrize('edges', (False, True))
 @pytest.mark.parametrize('texture', (False, True))
 def test_multiscale_basic_features(edges, texture):
     img = np.zeros((20, 20, 3))
     img[:10] = 1
     img += 0.05 * np.random.randn(*img.shape)
-    features = multiscale_basic_features(img, edges=edges, texture=texture, multichannel=True)
+    with expected_warnings(["'multichannel' is a deprecated argument"]):
```
Comment (stefanv) [inline skimage/feature/tests/test_basic_features.py:15]:
```suggestion
    with expected_warnings(["`multichannel` is a deprecated argument"]):
```

File: skimage/feature/tests/test_basic_features.py [hunk 6]
```diff
@@ -1,30 +1,70 @@
...
 
 
-def test_multiscale_basic_features_channel():
+def test_multiscale_basic_features_deprecated_multichannel():
     img = np.zeros((10, 10, 5))
     img[:10] = 1
     img += 0.05 * np.random.randn(*img.shape)
     n_sigmas = 2
-    features = multiscale_basic_features(img, sigma_min=1, sigma_max=2, multichannel=True)
+    with expected_warnings(["'multichannel' is a deprecated argument"]):
```
Comment (stefanv) [inline skimage/feature/tests/test_basic_features.py:30]:
```suggestion
    with expected_warnings(["`multichannel` is a deprecated argument"]):
```

File: skimage/feature/tests/test_basic_features.py [hunk 7]
```diff
@@ -1,30 +1,70 @@
...
+        features = multiscale_basic_features(img, sigma_min=1, sigma_max=2,
+                                             multichannel=True)
     assert features.shape[-1] == 5 * n_sigmas * 4
     assert features.shape[:-1] == img.shape[:-1]
+
+    # repeat prior test, but check for positional multichannel warning
+    with expected_warnings(["Providing the 'multichannel' argument"]):
```
Comment (stefanv) [inline skimage/feature/tests/test_basic_features.py:37]:
```suggestion
    with expected_warnings(["Providing the `multichannel` argument"]):
```

File: skimage/feature/tests/test_hog.py [hunk 8]
```diff
@@ -242,15 +242,37 @@ def test_hog_block_normalization_incorrect_error():
...
     img = data.astronaut()
     img[:, :, (1, 2)] = 0
-    hog_ref = feature.hog(img, multichannel=True, block_norm='L1')
+    with expected_warnings(["'multichannel' is a deprecated argument"]):
+        hog_ref = feature.hog(img, multichannel=True, block_norm='L1')
 
     for n in (1, 2):
-        hog_fact = feature.hog(np.roll(img, n, axis=2), multichannel=True,
-                               block_norm='L1')
+        with expected_warnings(["'multichannel' is a deprecated argument"]):
```
Comment (stefanv) [inline skimage/feature/tests/test_hog.py:256]:
```suggestion
        with expected_warnings(["`multichannel` is a deprecated argument"]):
```

File: skimage/feature/tests/test_hog.py [hunk 9]
```diff
@@ -242,15 +242,37 @@ def test_hog_block_normalization_incorrect_error():
...
+            hog_fact = feature.hog(np.roll(img, n, axis=2), multichannel=True,
+                                   block_norm='L1')
+        assert_almost_equal(hog_ref, hog_fact)
+
+        # repeat prior test, but check for positional multichannel warning
+        with expected_warnings(["Providing the 'multichannel' argument"]):
```
Comment (stefanv) [inline skimage/feature/tests/test_hog.py:262]:
```suggestion
        with expected_warnings(["Providing the `multichannel` argument"]):
```

**Review by stefanv — APPROVED** [review]
I highlighted some quote inconsistencies, but otherwise LGTM!

Thanks, Greg.

**Comment by grlee77** [comment]
Thanks, @stefan. I think I copy-pasted the same warning messages in the other PRs, so will update those with the same backtick fixes too.

**Pull request merged**